cow_instrument/InstrumentTree.h

```diff
@@ -17,30 +16,38 @@ class Detector;
  */
 class InstrumentTree {
 public:
-  InstrumentTree(std::unique_ptr<const Node> &&root, size_t nDetectors);
-
-  InstrumentTree(const InstrumentTree& other);
-
-  std::unique_ptr<NodeIterator> iterator() const;
+  InstrumentTree(std::vector<Node> &&nodes, size_t nDetectors);
+  InstrumentTree(const InstrumentTree& original) = default;
 
   const Node &root() const;
 
   const Detector &getDetector(size_t detectorIndex) const;
 
   unsigned int version() const;
 
-  std::unique_ptr<const InstrumentTree> modify(const Command &command) const;
+  InstrumentTree modify(size_t node, const Command &command) const;
+  InstrumentTree modify(const Node *node, const Command &command) const;
```

Comment:
That's a good question. I'm not sure which access route is going to be the most feasible. The fact that the `InstrumentTree` stores a flattened structure is essentially private. That would imply that knowing the index you desire should be somewhat abstracted. To obtain the `Node*` you probably would like to get this via some kind of `TreeIterator` such that you could for example "Get Node called some-surface-holder". I don't have a complete answer to this yet. Happy to discuss.